tools/sof-fw-boot-test.py

```diff
@@ -0,0 +1,135 @@
+    # put the DSP in D3
+    debugfs_entry = debugfs_path + "/dsp_power_state"
+    with open(debugfs_entry, "w", encoding="utf8") as dbgf:
+        dbgf.write("D3\n")
+
+    # check if the DSP is in D3
+    with open(debugfs_entry, "r", encoding="utf8") as dbgf:
+        power_state = dbgf.read(1024)
+        if power_state != "D3":
+            sys.exit("Failed booting firmware. DSP is not in D3")
+
+    # unload current firmware
+    debugfs_entry = debugfs_path + "/unload_fw"
+    with open(debugfs_entry, "w", encoding="utf8") as dbgf:
+        dbgf.write("1\n")
+
+    # get current fw_state and continue to boot only if the current state is 'PREPARE'
+    debugfs_entry = debugfs_path + "/fw_state"
+    with open(debugfs_entry, "r", encoding="utf8") as dbgf:
+        fw_state = dbgf.read(1024)
+        fw_state = fw_state.rstrip()
+        if fw_state != "PREPARE":
+            sys.exit("Cannot boot firmware boot from current state {fw_state}")
+
+    # load and boot firmware
+    debugfs_entry = debugfs_path + "/boot_fw"
+    start = time.time()
+    with open(debugfs_entry, "w", encoding="utf8") as dbgf:
+        dbgf.write("1\n")
+    end = time.time()
+
+    # get current fw_state
+    debugfs_entry = debugfs_path + "/fw_state"
+    with open(debugfs_entry, "r", encoding="utf8") as dbgf:
+        fw_state = dbgf.read(1024)
+        fw_state = fw_state.rstrip()
+
+    # calculate boot time
+    boot_time_ms = round((end - start) * 1000, 3)
+
+    return boot_time_ms, fw_state
+
+
+def main():
+    """Main function for stress testing"""
+    cmd_args = parse_cmdline()
+
+    # Get firmware file path
+    fw_path = cmd_args["fw_path"]
+
+    # Get firmware file name
+    fw_filename = cmd_args["firmware"]
+
+    num_iter = cmd_args["iter"]
+    output = f"""==============================================================================
+	Starting boot stress test with:
+	Firmware filename: {fw_filename}
+	Path to firmware file: {fw_path}
+	Number of Iterations: {num_iter}
+=============================================================================="""
+    logging.info(output)
+
+    # set the firmware filename & path
+    with open(debugfs_path + "/fw_filename", "w", encoding="utf8") as dbgf:
+        dbgf.write(fw_filename + "\n")
+    with open(debugfs_path + "/fw_path", "w", encoding="utf8") as dbgf:
+        dbgf.write(fw_path + "\n")
+
+    total_boot_time_ms = 0
+    min_boot_time_ms = sys.maxsize
+    max_boot_time_ms = 0
+
+    for i in range(num_iter):
+        boot_time_ms, fw_state = boot_fw()
+        # check if fw_state is COMPLETE
+        if fw_state != "COMPLETE":
+            sys.exit(f"Firmware boot failed at iteration {i}")
+
+        total_boot_time_ms += boot_time_ms
+        min_boot_time_ms = min(min_boot_time_ms, boot_time_ms)
+        max_boot_time_ms = max(max_boot_time_ms, boot_time_ms)
+
+        logging.info(f"Firmware boot iteration {i} completed in {boot_time_ms} ms")
```

Comment:
```suggestion
        logging.info("Firmware boot iteration %d completed in %s ms" % (i, boot_time_ms))
```
To silence this LAST (!) pylint warning:
```
tools/sof-fw-boot-test.py:123:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
```

The warning is about uselessly formatting the string and "computing" the arguments  when the debug level is too low.